### DISCUSSION TITLE
ERROR: dependency 'unigd' is not available for package 'httpgd'

### DISCUSSION THREAD

**Discussion by aminuldu07** [nx10/httpgd#157] [Q&A]
HI,
I am trying to install 'httpgd' where my R version is 4.2.2
However, I am getting this errored message. 
ERROR: dependency 'unigd' is not available for package 'httpgd'. 
Can you please help me in this regard.

**Comment by eitsupi** [comment]
What is the URL of CRAN?
That is odd because unigd is on CRAN.
https://cran.r-project.org/package=unigd

**Reply by aminuldu07** [reply]
while unidg is on CRAN but may be it has some bug. i tried and got this error 
C:\rtools42\x86_64-w64-mingw32.static.posix\bin/ld.exe: cannot find -lsharpyuv
collect2.exe: error: ld returned 1 exit status
no DLL was created
ERROR: compilation failed for package 'unigd'

" i ASKED my friend to try and he got this erroor, while rtools42 rtool42 is seems ok 
installing *source* package 'unigd' ...
** using staged installation
** libs

*** arch - i386
"C:/PROGRA~1/R/R-41~1.2/bin/i386/Rscript.exe" "../tools/winlibs.R" 2.7.4 & \
"C:/PROGRA~1/R/R-41~1.2/bin/i386/Rscript.exe" "../tools/wincairo.R" 1.16.0
"C:/rtools40/mingw32/bin/"g++  -std=gnu++14 -I"C:/PROGRA~1/R/R-41~1.2/include" -DNDEBUG -Ilib -I../inst/include -I../windows/harfbuzz-2.7.4/include -I../windows/cairo-1.16.0/include/cairo -I../windows/cairo-1.16.0/include/freetype2 -DFMT_HEADER_ONLY -I'C:/Users/Md.Ali/R_library_4.1.2/cpp11/include' -I'C:/Program Files/R/R-4.1.2/library/systemfonts/include'        -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign  -c base_64.cpp -o base_64.o
base_64.cpp:7:10: fatal error: png.h: No such file or directory
 #include <png.h>
          ^~~~~~~
compilation terminated.
make: *** [C:/PROGRA~1/R/R-41~1.2/etc/i386/Makeconf:245: base_64.o] Error 1
ERROR: compilation failed for package 'unigd'
* removing 'C:/Users/Md.Ali/R_library_4.1.2/unigd'
Warning message:
In i.p(...) :
  installation of package ‘c:/temp/1/RtmpywpD37/file21504e76781b/unigd_0.1.0.tar.gz’ had non-zero exit status

**Reply by eitsupi** [reply]
I recommend trying the latest version of R.
Isn't it possible that the tools included in Rtools on Windows are too old?
